common/dpcpp/slab_hash.hpp

```diff
@@ -1,182 +1,183 @@
 #pragma once
 
+#include "dpcpp_common.hpp"
 #include <CL/sycl.hpp>
 #include <algorithm>
-#include "dpcpp_common.hpp"
-
-#define SUBGROUP_SIZE 16
-#define CONST 64
-#define SLAB_SIZE CONST * SUBGROUP_SIZE
+#include <optional>
 
-#define CLUSTER_SIZE 1024
+constexpr size_t SUBGROUP_SIZE = 16;
+constexpr size_t CONST = 64;
+constexpr size_t SLAB_SIZE = CONST * SUBGROUP_SIZE;
 
-#define BUCKETS_COUNT 128
+constexpr size_t CLUSTER_SIZE = 1024;
 
-#define EMPTY_UINT32_T 4294967295
+constexpr size_t BUCKETS_COUNT = 128;
 
-using std::pair;
+constexpr size_t EMPTY_UINT32_T = std::numeric_limits<uint32_t>::max();
 
-template <size_t A, size_t B, size_t P>
-struct Hasher {
-    size_t operator()(const uint32_t &k) { return ((A * k + B) % P) % BUCKETS_COUNT; };
+namespace SlabHashHashers {
```

Comment:
Well, I meant to wrap the whole thing in a namespace :)

Reply:
oops :)